cpp/doxygen/.clang-format

```diff
@@ -0,0 +1,28 @@
+---
```

Comment:
It's not clear to me why we need this .clang-format file.

Is it different from the top-level .clang-format file?

Reply:
The only difference is the line length, I pick 90 for the examples to avoid horizontal scrolling.

Reply:
Reworked the config file to use `BasedOnStyle: InheritParentConfig` and only include the `ColumnLimit`